### PR TITLE
63: Repository-wide .gitignore

### LINKED ISSUE #63: Repository-wide .gitignore
The existing .gitignore file should be brought up to the root directory and updated. A good reference: https://github.com/github/gitignore

### PR DESCRIPTION
**Pull request by Subzeero**
Closes #63

# What was the issue
Our `.gitignore` file was located in a subdirectory and was missing some necessary files to ignore.

# What was done and why
Moved the `.gitignore` file up to the repository root and added some common templates for Node.js, macOS, Windows, and JetBrains IDEs.

# How it was tested
N/A

### CODE REVIEW

**Review by matthew-buglass — APPROVED** [review]

**Review by trangnguyen3010 — CHANGES_REQUESTED** [review]
- [x] Prefix the commit with the issue number

**Comment by Subzeero** [comment]
Hey guys, I was reviewing the a support pages from JetBrains and there are some files in the .idea folder that are supposed to be shared with Git. I have added these in the most recent push. These files are "needed" to be shared, so there shouldn't be any issues, but please check and let me know.

**Review by matthew-buglass — APPROVED** [review]
As this gets merged in, I would recommend letting everyone know the their local environment might change a bit with the .idea files being committed.

**Comment by trangnguyen3010** [comment]
I checked out the branch and have not seen any problems so far! 👍

**Review by trangnguyen3010 — APPROVED** [review]

File: .idea/inspectionProfiles/Project_Default.xml
[the remark this answers is not in the document]
Comment (matthew-buglass) [reply .idea/inspectionProfiles/Project_Default.xml:1]: This acts as the default setting for the project. The idea files that are not ignored allow for a standard environment across all developers to assist in consistency.

File: .gitignore
[the remark this answers is not in the document]
Comment (Subzeero) [reply .gitignore:1]: My understanding is that these files should be ignored everywhere. If it is not a direct file path, it should catchall.

**Pull request merged**